### ISSUE TITLE
Create buildings from OS open Zoomstack data

### ISSUE THREAD

**Issue by Robinlovelace** [cyipt/actdev#88]
As discussed with @mem48 

https://osdatahub.os.uk/downloads/open

And https://www.wisemover.co.uk/carbon/os.html

Could reduce need for procgen very cool.

**Comment by mem48** [comment]
The version I hosted doesn't include the full zoom. The max zoom includes building polygons

![image](https://user-images.githubusercontent.com/15819577/107978963-99800600-6fb5-11eb-88eb-68c802cf5e9c.png)

**Comment by dabreegster** [comment]
Very neat! Ideally these could be imported into OSM, if the license allows for it. I found https://wiki.openstreetmap.org/wiki/Ordnance_Survey_OpenData, but haven't dug in.

**Comment by mem48** [comment]
You can download it as a GeoPackage and extract the buildings_local layer

**Comment by mem48** [comment]
@dabreegster Its open government licence data, but I think the OSM community doesn't want to use it as they are low-res polygons. good for the base map but not as good as the properly traced buildings in OSM. But for places where OSM has no buildings are better than nothing.

**Comment by dabreegster** [comment]
@mem48 Thanks for clarifying why it's not in OSM! I should be able to parse the GeoPackage and use to augment what's in OSM instead of doing the procedural generation. But as Robin prudently labeled, not a priority yet.

**Comment by mvl22** [comment]
Ah very good point! I'd completely forgotten about these. It should be pretty good data (though as you say, imperfect geometries).

Ironically these have been in the alpha app all along, in the form of the "OS outdoor" layer, which gives 3D buildings...

Presumably ogr2ogr can convert the GeoPackage layer and convert to GeoJSON.

**Comment by mem48** [comment]
@mvl22 yes, or I could extract it as a geojson for you if that would help.
